configure

```diff
@@ -0,0 +1,265 @@
+#!/bin/sh
+#
+# Configure Themis build.
+#
+# Normally you should not need to call this script explicitly as "make"
+# executes it automatically if needed. But by all means, you can do
+#
+#     ./configure
+#     make
+#     sudo make install
+#
+# to install Themis in a traditional way, or if you need to configure
+# something non-standard.
+#
+# Important environment variables:
+#
+#     BUILD_PATH - path to build directory (default: "build")
+#
+# ./configure expects to be run from the source root with build directory
+# configured by BUILD_PATH if necessary.
+#
+# Run
+#
+#     ./configure --help
+#
+# to see available options.
+
+set -eu
+
+BUILD_PATH=${BUILD_PATH:-build}
+
+CC=${CC:-cc}
+CMAKE=${CMAKE:-cmake}
+
+usage() {
+    cat <<EOF
+Usage:
+    $0 [options...] [<var>=<value>...]
+
+Options:
+    --prefix=path           installation prefix
+    --bindir=path           installation path for binaries
+    --includedir=path       installation path for headers
+    --libdir=path           installation path for shared libraries
+    --jnidir=path           installation path for JNI libraries
+    --pkgconfigdir=path     installation path for pkg-config files
+
+    --help                  read this help
+EOF
+}
+
+die() {
+    exec >&2
+    echo $@
+    echo
+    usage
+    exit 1
+}
+
+configure_mk="$BUILD_PATH/configure.mk"
+
+write_boilerplate() {
+    mkdir -p "$(dirname "$configure_mk")"
+    cat > "$configure_mk"  <<EOF
+# DO NOT EDIT
+#
+# Generated automatically by $0
+
+EOF
+}
+
+set_variable() {
+    eval "$1=\"$2\""
+    echo "$1 = $2" >> "$configure_mk"
+}
+
+append_variable() {
+    eval "$1=\"$1 $2\""
+    echo "$1 += $2" >> "$configure_mk"
+}
+
+standard_dirs="prefix bindir includedir libdir jnidir pkgconfigdir"
+
+is_standard_dir() {
+    for dir in $standard_dirs
+    do
+        if [ "$1" = "$dir" ]
+        then
+            return 0
+        fi
+    done
+    return 1
+}
+
+parse_variable_override() {
+    case "$1" in
+      --*=*)
+        var=${1#--}
+        value=${var#*=}
+        var=${var%%=*}
```

Comment:
oh, that is why I hate bash and makefile languages... too much non-alpha symbols in syntax

Reply:
You'll enjoy APL then 😉

> {↑1 ⍵∨.∧3 4=+/,¯1 0 1∘.⊖¯1 0 1∘.⌽⊂⍵}